sensor/common/sensor/central_communication_test.go

```diff
@@ -84,70 +64,78 @@ func Test_CentralCommunicationSuite(t *testing.T) {
-	comms.Start(c.conn, &reachable, c.mockHandler, c.mockDetector)
+var _ central.SensorServiceClient = (*MockSensorServiceClient)(nil)
 
-	return componentMessages
+var centralSyncMessages = []*central.MsgToSensor{
+	debuggerMessage.SensorHello("00000000-0000-4000-A000-000000000000"),
+	debuggerMessage.ClusterConfig(),
+	debuggerMessage.PolicySync([]*storage.Policy{}),
+	debuggerMessage.BaselineSync([]*storage.ProcessBaseline{}),
 }
 
-func createConnectionAndStartServer(fakeCentral *centralDebug.FakeService) (*grpc.ClientConn, func()) {
-	buffer := 1024 * 1024
-	listener := bufconn.Listen(buffer)
-
-	fakeCentral.ServerPointer = grpc.NewServer()
-	central.RegisterSensorServiceServer(fakeCentral.ServerPointer, fakeCentral)
-
-	go func() {
-		utils.IgnoreError(func() error {
-			return fakeCentral.ServerPointer.Serve(listener)
-		})
-	}()
+func (c *centralCommunicationSuite) Test_StartCentralCommunication() {
+	// Create a fake SensorComponent
+	responsesC := make(chan *message.ExpiringMessage)
+	defer close(responsesC)
+	comm := NewCentralCommunication(false, NewFakeSensorComponent(responsesC))
 
-	conn, err := grpc.DialContext(context.Background(), "", grpc.WithContextDialer(func(ctx context.Context, s string) (net.Conn, error) {
-		return listener.Dial()
-	}), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	reachable := concurrency.Flag{}
+	mockService := &MockSensorServiceClient{
+		connected: concurrency.NewSignal(),
+		client:    mocksClient.NewMockServiceCommunicateClient(c.controller),
+	}
 
-	if err != nil {
-		panic(err)
+	expectSyncMessages(centralSyncMessages, mockService)
+	ch := make(chan struct{})
+	mockService.client.EXPECT().Send(gomock.Any()).Times(1).DoAndReturn(func(msg *central.MsgFromSensor) error {
+		defer close(ch)
+		c.Assert().NotNil(msg.GetEvent().GetSynced())
+		return nil
+	})
+	// Start the go routine with the mocked client
+	go comm.(*centralCommunicationImpl).sendEvents(mockService, &reachable, c.mockHandler, c.mockDetector)
+	mockService.connected.Wait()
+	// Pretend that a component (listener) is sending the sync event
+	responsesC <- message.New(syncMessage())
+	select {
+	case <-ch:
+		break
+	case <-time.After(5 * time.Second):
+		c.Fail("timeout reached waiting for the sync event")
 	}
+}
 
-	closeF := func() {
-		utils.IgnoreError(listener.Close)
-		fakeCentral.ServerPointer.Stop()
+func expectSyncMessages(messages []*central.MsgToSensor, service *MockSensorServiceClient) {
+	key := strings.ToLower(centralsensor.SensorHelloMetadataKey)
+	md := metadata.MD{
+		key: []string{"true"},
+	}
+	service.client.EXPECT().Header().AnyTimes().Return(md, nil)
+	service.client.EXPECT().CloseSend().AnyTimes()
+	service.client.EXPECT().Send(gomock.Any()).Return(nil)
+	service.client.EXPECT().Context().AnyTimes().Return(context.Background())
+	var orderedCalls []*gomock.Call
+	for _, m := range messages {
+		orderedCalls = append(orderedCalls, service.client.EXPECT().Recv().Return(m, nil))
```

Comment:
nit: I think this might make the assertion a bit more robust. I'm not 100% sure if it makes a difference, but I changed locally and the tests are still passing.  
```suggestion
		orderedCalls = append(orderedCalls, service.client.EXPECT().Recv().Times(1).Return(m, nil))
```

Reply:
I added `Time(1)` to be more explicit but I think it is redundant to have it (see the default values [here](https://github.com/golang/mock/blob/aba2ff9a6844d5e3289e8472d3217d5b3090f083/gomock/call.go#L79))